src/main/java/com/gluonhq/richtext/viewmodel/RichTextAreaViewModel.java

```diff
@@ -76,8 +88,9 @@ protected void invalidated() {
             if (!hasSelection()) {
                 Decoration decorationAtCaret = getTextBuffer().getDecorationAtCaret(get());
                 if (decorationAtCaret instanceof TextDecoration) {
-                    setDecoration(decorationAtCaret);
+                    setDecorationAtCaret(decorationAtCaret);
                 }
+                getParagraphWithCaret().ifPresent(p -> setDecorationAtParagraph(p.getDecoration()));
```

Comment:
Since TextDecoration and ParagraphDecoration are mutually exclusive, the order doesn't matter. Is my understanding correct?

Reply:
yes, indeed, you can't apply both at the same time, so order doesn't matter.

Reply:
Preset paragraph style (#76) is currently implemented (#124) using TextDecorations. Do you see the implementation getting changed because of this PR?

Reply:
Paragraph styles should be actually a combination of TextDecoration and ParagraphDecoration, I don't see a problem with that so far.